### PR TITLE
Merge develop into 2.0

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

**Review by abdulmth — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -117,83 +117,15 @@ iota-sdk = { git = "https://github.com/iotaledger/iota-sdk", branch = "develop"
...
-        .finish()
-        .await?;
-            
-    let info = client.get_info().await?;
-    println!("Node Info: {info:?}")
-    
-    Ok(())
-}
-```
+The following example creates a Client instance connected to the Shimmer Testnet, and retrieves the node's information by calling `Client.get_info()`, and then print the node's information.
```
Comment (DaughterOfMars) [inline README.md:120]: Feels like this was a mistaken revert

File: sdk/examples/wallet/getting_started.rs [hunk 2]
```diff
@@ -21,45 +19,38 @@ use iota_sdk::{
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    // This example uses secrets in environment variables for simplicity which should not be done in production.
-    dotenvy::dotenv().ok();
```
Comment (DaughterOfMars) [inline sdk/examples/wallet/getting_started.rs:25]: We don't use the ENV anymore?

File: README.md (hunk 1 above)
Comment (thibault-martinez) [reply README.md:120]: No that's fine

Comment (thibault-martinez) [reply README.md:120]: Also there was no conflict on the README

File: sdk/src/client/node_api/core/routes.rs [hunk 3]
```diff
@@ -201,8 +166,10 @@ impl ClientInner {
     /// POST /api/core/v3/blocks
     pub async fn post_block_raw(&self, block: &Block) -> Result<BlockId> {
         const PATH: &str = "api/core/v3/blocks";
+
         let timeout = self.get_timeout().await;
 
+        // TODO add Client::post_request_bytes
```
Comment (DaughterOfMars) [inline sdk/src/client/node_api/core/routes.rs:172]: Not needed

File: sdk/examples/wallet/getting_started.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/examples/wallet/getting_started.rs:25]: This is a getting_started example so better not

File: sdk/src/client/node_api/core/routes.rs (hunk 3 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:172]: Why?

Comment (DaughterOfMars) [reply sdk/src/client/node_api/core/routes.rs:172]: NVM LOL

File: README.md (hunk 1 above)
Comment (DaughterOfMars) [reply README.md:120]: Okie dokie 🤷‍♀️

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**